src/lib/pipeline.ml

```diff
@@ -545,6 +549,67 @@ module Compiler = struct
           ~name:(sprintf "single-end %s"
                    (Filename.basename r1#product#path))
     in
+    let parallelize_alignment ~make_aligner sample =
+      match sample with
+      | Paired_end_sample (name,
+                           Gunzip_concat r1_list,
+                           Gunzip_concat r2_list) ->
+        let exploded =
+          let count = ref 0 in
+          List.map2 r1_list r2_list
+            ~f:(fun (Fastq_gz wf1 as r1) (Fastq_gz wf2 as r2) ->
+                let new_name =
+                  incr count; sprintf "%s-fragment-%d" name !count in
+                make_aligner (
+                  Paired_end_sample (new_name,
+                                     Gunzip_concat [r1],
+                                     Gunzip_concat [r2]))) in
+        let bams = List.map exploded ~f:(compile_aligner_step ~compiler) in
+        Samtools.merge_bams ~run_with:machine bams
+          (result_prefix ^ "-merged.bam")
+      | other ->
+        failwith "parallelize_alignment: Not fully implemented"
+    in
+    let catch_parallelize_aligner aligner sample =
+      let option =
+        List.find_map compiler.options
+          (function
+          | `Parallel_alignment_over_fastq_fragments (al, todo)
+            when List.mem ~set:al aligner -> Some todo
+          | _ -> None)
+      in
+      match sample with
+      | Paired_end_sample (name,
+                           Gunzip_concat r1_list,
+                           Gunzip_concat r2_list)
+        when List.length r1_list > 1 ->
+        begin match option with
+        | Some _ -> `Caught
+        | None -> `Not_caught
+        end
+      | Paired_end_sample (name,
+                           Gunzip_concat r1_list,
+                           Gunzip_concat r2_list)
+        when List.length r1_list = 1 -> `Not_caught
+      | other ->
+        begin match option with
+        | Some `Silent
+        | None -> `Not_caught
```

Comment:
I think the default of when a compiler optimization cannot be made, but is expected to by the user, should be noisy alert, not silent failure and continuing with the compilation.